src/main/java/com/theelixrlabs/healthcare/controller/DoctorController.java

```diff
@@ -0,0 +1,35 @@
+package com.theelixrlabs.healthcare.controller;
+
+import com.theelixrlabs.healthcare.constants.DoctorConstants;
+import com.theelixrlabs.healthcare.service.DoctorService;
+import com.theelixrlabs.healthcare.dto.DoctorDto;
+import com.theelixrlabs.healthcare.response.SuccessResponse;
+import jakarta.validation.Valid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * Rest Controller for handling HTTP request and response.
+ */
+@RestController
+public class DoctorController {
+
+    @Autowired
+    DoctorService doctorService;
```

Comment:
please make it private

Reply:
Yes I removed @Autowired and implemented constructor based injection.